src/main/java/com/laytonsmith/core/functions/EventBinding.java

```diff
@@ -233,16 +246,122 @@ public boolean useSpecialExec() {
+											+ " is not a valid event identifier."
+											+ " It cannot match the regex \".*?:\\d*?\".", children.get(1).getTarget()));
+							}
+						} else if(children.get(0).getData().val().equals("priority")
+								&& children.get(1).getData().isInstanceOf(CString.TYPE)) {
+							try {
+								BoundEvent.Priority.valueOf(children.get(1).getData().val());
+							} catch (IllegalArgumentException ex) {
+								exceptions.add(new ConfigCompileException(children.get(1).getData().val()
+											+ " is not a valid enum in ms.lang.Priority",
+												children.get(1).getTarget()));
+							}
+						} else {
+							List<String> validProps = Arrays.asList("id", "priority");
+							String prop = children.get(0).getData().val();
+							if(!validProps.contains(prop)) {
+								env.getEnv(CompilerEnvironment.class).addCompilerWarning(fileOptions,
+											new CompilerWarning("Unexpected entry, this will be ignored.",
+													children.get(0).getTarget(), null));
+							}
+						}
+					}
+				}
+			}
+
+			// Validate prefilters, if we can
+			Event ev = EventList.getEvent(topChildren.get(0).getData().val());
+			if(ev != null) {
+				Event.PrefilterBuilder prefilterBuilder = ev.getPrefilters();
+				if(prefilterBuilder != null) {
+					Map<String, Pair<Prefilters.PrefilterType, Event.PrefilterMatcher>> prefilters = prefilterBuilder.build();
+					child = topChildren.get(2);
+					if(child.getData() instanceof CFunction && child.getData().val().equals("array")) {
+						for(ParseTree node : child.getChildren()) {
+							if(node.getData() instanceof CFunction && node.getData().val().equals("centry")) {
+								List<ParseTree> children = node.getChildren();
+								if(prefilters.containsKey(children.get(0).getData().val())) {
+									ParseTree value = children.get(1);
+									CClassType valType = value.getType(env);
+									Prefilters.PrefilterType type = prefilters.get(children.get(0).getData().val()).getKey();
+									String warn = null;
+									switch(type) {
+										case BOOLEAN_MATCH:
+											if(!valType.unsafeDoesExtend(CBoolean.TYPE)) {
+												warn = "Expected a boolean here, this may not perform as expected.";
+											}
+											break;
+										case LOCATION_MATCH:
+											if(!valType.unsafeDoesExtend(CArray.TYPE)) {
+												warn = "Expecting an array here.";
+											}
+											break;
+										case REGEX:
+											if(!valType.unsafeDoesExtend(CString.TYPE)) {
+												warn = "Expecting a string (regex) type here.";
+											} else if(value.isConst()) {
+												try {
+													Pattern.compile(value.getData().val());
+												} catch (PatternSyntaxException ex) {
+													exceptions.add(new ConfigCompileException(ex.getMessage(), t));
+												}
+											}
+											break;
+										case STRING_MATCH:
+											if(!valType.unsafeDoesExtend(CString.TYPE)) {
```

Comment:
Gets "ClassNotFoundException: auto is not a real type" when using a function with a return type of auto.